services/base.js

```diff
@@ -116,28 +154,37 @@ class BaseService {
         {
           title,
           query,
+          namedParams,
           exampleUrl,
           previewUrl,
+          pattern,
           urlPattern,
           staticExample,
           documentation,
           keywords,
         },
         index
       ) => {
+        pattern = pattern || urlPattern || this.route.pattern
```

Comment:
I'm confused about why we want both `pattern` and `urlPattern` and what situation we use each one in. Can you explain the difference.

Reply:
We wouldn't want both. It's a temporary alias to `pattern` to make this diff smaller. I added it to the docstring. Does that make sense?

Reply:
> Does that make sense?

Maaaybe.. Am I right that essentially what you really want to do is rename `urlPattern` to `pattern`, but we're supporting both so that we don't have to change every single badge that declares `urlPattern` to `pattern` in this PR.. but then we want to do that at a later date and then remove `urlPattern` when we've done that?